packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -3473,22 +3485,31 @@ function commitRootImpl(
   }
 
   if (recoverableErrors !== null) {
-    // There were errors during this render, but recovered from them without
-    // needing to surface it to the UI. We log them here.
-    const onRecoverableError = root.onRecoverableError;
-    for (let i = 0; i < recoverableErrors.length; i++) {
-      const recoverableError = recoverableErrors[i];
-      const errorInfo = makeErrorInfo(recoverableError.stack);
-      if (__DEV__) {
-        runWithFiberInDEV(
-          recoverableError.source,
-          onRecoverableError,
-          recoverableError.value,
-          errorInfo,
-        );
-      } else {
-        onRecoverableError(recoverableError.value, errorInfo);
+    const prevTransition = ReactSharedInternals.T;
+    const previousUpdateLanePriority = getCurrentUpdatePriority();
+    setCurrentUpdatePriority(DiscreteEventPriority);
+    ReactSharedInternals.T = null;
```

Comment:
Why do we need to set the priority here? Is this just so the reactor is more 1-1?

Reply:
Yea it's mainly to keep it 1-1. Since this is a user space callback you could have a setState in here such as updating an error dialog.

Therefore this must have some semantic. It used to be Discrete based on how it was factored but it makes sense that it would be.

Whether it always should be anyway is an interesting question though. It really shouldn't be possible to run a commit inside a startTransition so I'm not sure how it could ever be anything else.